### PR TITLE
Steam native remove

### PR DESCRIPTION
**Pull request by jonringer**
###### Motivation for this change
Trying to split out #157907 into smaller fixes

To test:
```
nix run --impure github:jonringer/nixpkgs/steam-native-remove#steam
```
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -1143,6 +1143,7 @@ mapAliases ({
   squid4 = squid;  # added 2019-08-22
   sshfsFuse = throw "'sshfsFuse' has been renamed to/replaced by 'sshfs-fuse'"; # Converted to throw 2022-02-22
   stanchion = throw "Stanchion was part of riak-cs which is not maintained anymore"; # added 2020-10-14
+  steam-run-native = steam-run; # added 2022-02-21
```
Comment (Artturin) [inline pkgs/top-level/aliases.nix:1146]: I think it would be better to make this into a throw since the package was already broken

Comment (jonringer) [reply pkgs/top-level/aliases.nix:1146]: This is what I had originally, but I don't feel strongly either way.

Comment (jonringer) [reply pkgs/top-level/aliases.nix:1146]: I don't feel super strongly either way, but it would probably be a better user experience to have an alias

**Comment by jonringer** [comment]
```
https://github.com/NixOS/nixpkgs/pull/161997

1 package marked as broken and skipped:
steam-run-native

13 packages built:
bottles itch itch-setup kodiPackages.steam-launcher lutris minigalaxy playonlinux protontricks steam steam-run steam-tui steamcmd wpsoffice
```

**Comment by Atemu** [comment]
This broke Steam for me, it's now unable to locate its state dir.

**Comment by jonringer** [comment]
> This broke Steam for me, it's now unable to locate its state dir.

Like the game directory? I was able to access my library fine

File: pkgs/games/steam/fhsenv.nix [hunk 2]
```diff
@@ -51,7 +50,6 @@ let
   # Zachtronics and a few other studios expect STEAM_LD_LIBRARY_PATH to be present
   exportLDPath = ''
     export LD_LIBRARY_PATH=${lib.concatStringsSep ":" ldPath}''${LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH
-    export STEAM_LD_LIBRARY_PATH="$STEAM_LD_LIBRARY_PATH''${STEAM_LD_LIBRARY_PATH:+:}$LD_LIBRARY_PATH"
```
Comment (Atemu) [inline pkgs/games/steam/fhsenv.nix:54]: Reversing this deletion makes it work again.

Comment (jonringer) [reply pkgs/games/steam/fhsenv.nix:54]: Interesting, I'm fine with reverting this change

File: pkgs/games/steam/fhsenv.nix [hunk 3]
```diff
@@ -163,60 +141,44 @@ in buildFHSUserEnv rec {
     xorg.libxshmfence
     xorg.libXxf86vm
     libelf
-  ] ++ (if (!nativeOnly) then [
-    (steamPackages.steam-runtime-wrapped.override {
-      inherit runtimeOnly;
-    })
-  ] else [
```
Comment (Atemu) [inline pkgs/games/steam/fhsenv.nix:170]: Have a closer look at this if expression; everything below this are the *packages for nativeOnly*.

This change makes it so nativeOnly packages are always included. That's not what we want!

Comment (Atemu) [reply pkgs/games/steam/fhsenv.nix:170]: Deleting all of the nativeOnly packages again fixes my issue.

Comment (jonringer) [reply pkgs/games/steam/fhsenv.nix:170]: you can see how many duplicates we had, more and more launchers were requiring additional dependencies

Comment (Atemu) [reply pkgs/games/steam/fhsenv.nix:170]: Also, do we not need the runtime? It seems to function without but I wonder why...

Comment (jonringer) [reply pkgs/games/steam/fhsenv.nix:170]: it's not really used, steam will manage it's own runtimes, the one's packaged within nix are considered bootstrap runtimes, and they are just there to make sure it launches (IIRC)

Comment (Atemu) [reply pkgs/games/steam/fhsenv.nix:170]: Shouldn't we strive to make nativeOnly a thing then? If all we need is bootstrap for Steam to launch its own runtime, we don't need to package and distribute a huge runtime of Ubuntu packages.

**Comment by Atemu** [comment]
> > This broke Steam for me, it's now unable to locate its state dir.
> 
> Like the game directory? I was able to access my library fine

Sorry, missed your reply. It can't find the Steam directory:

    Couldn't find Steam content, did you move it?

My case is a bit special as my Steam directory is a symlink to somewhere else, that might be why?

File: pkgs/games/steam/fhsenv.nix (hunk 3 above)
Comment (jonringer) [reply pkgs/games/steam/fhsenv.nix:170]: It has been broken for years

**Comment by jonringer** [comment]
where is the symlink? It's likely not being mounted

**Comment by Atemu** [comment]
It's at `/var/opt/games` but that's not it. The Steam from the parent can find it just fine and nothing changed about the mounting logic. 

I found the culprit: `networkmanager098`

Why tf would that interfere with Steam finding its library?! What is it even there for?

**Comment by Atemu** [comment]
@abbradar you added it a long time ago, do you know why?

**Comment by Artturin** [comment]
i'll submit a pr that removes that ancient(2015) networkmanager version

**Comment by Atemu** [comment]
I'm still a bit uncertain about all of those packages added in this PR. Do we really need them?

I guess closure size isn't really a concern with Steam (only went up by a few %) but it's not ideal either.

I'd be fine with removing nm098 and leaving things as they are until we strip away our steamrt where we'll have to re-evaluate all of these dependencies anyways.

File: pkgs/games/steam/fhsenv.nix (hunk 3 above)
Comment (Atemu) [reply pkgs/games/steam/fhsenv.nix:170]: I know but that was before Steam did its own sandboxing.

File: doc/builders/packages/steam.section.md [hunk 4]
```diff
@@ -56,7 +56,7 @@ Use `programs.steam.enable = true;` if you want to add steam to systemPackages a
 
 ## steam-run {#sec-steam-run}
 
-The FHS-compatible chroot used for Steam can also be used to run other Linux games that expect a FHS environment. To use it, install the `steam-run-native` package and run the game with
+The FHS-compatible chroot used for Steam can also be used to run other Linux games that expect a FHS environment. To use it, install the `steam-run` package and run the game with
```
Comment (McSinyx) [inline doc/builders/packages/steam.section.md:59]: Since the non-native is non-free, is there any alternative for programs expecting an FHS environment?

**Comment by Atemu** [comment]
I don't think that should be related to steam-run, it should be a new package. `fhs-run` or something.

**Comment by abbradar** [comment]
Regarding `networkmanager098` -- I faintly remember it was needed for Steam itself. If we drop native Steam Runtime support, it's most definitely not needed.

I'd say we should try to keep the same packages as exposed by Steam Runtime. As it's a de facto standard for games runtime right now, this would improve chances that games from GOG and other platforms run smoothly. I don't expect anything to want old NetworkManager version specifically, but I would leave most other libraries intact.

Regarding `fhs-run` -- yes, that's a fine name! We could mention in the documentation that `steam-run` is non-free, and `fhs-run` is the default way to go. I'm also not sure our packaged Steam Runtime (and by extent `steam-run`) will be here for long (see discussion in https://github.com/ValveSoftware/Dota-2/issues/2018), but that's out of scope.

File: doc/builders/packages/steam.section.md (hunk 4 above)
Comment (fgaz) [reply doc/builders/packages/steam.section.md:59]: +1! I don't understand why this was removed. it was very useful to me. programs like `minigalaxy` shouldn't require unfree dependencies

Comment (jonringer) [reply doc/builders/packages/steam.section.md:59]: We probably shouldn't be using a steam runtime for `minigalaxy`, steam itself is unfree, so it's expected to need unfree to use steam (or downstream package)

**Comment by jonringer** [comment]
I agree with @Atemu, we should probably have a solution which isn't steam specific.

Actually, we should probably refactor the steamfhs.nix to use the generic solution + the steam specific options

**Pull request merged**